subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/java/KotlinAnalysisSourceRootsExtractor.kt

```diff
@@ -0,0 +1,26 @@
+package org.jetbrains.dokka.analysis.kotlin.descriptors.compiler.java
+
+import org.jetbrains.dokka.DokkaConfiguration
+import org.jetbrains.dokka.analysis.java.SourceRootsExtractor
+import org.jetbrains.dokka.analysis.kotlin.descriptors.compiler.CompilerDescriptorAnalysisPlugin
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.plugability.plugin
+import org.jetbrains.dokka.plugability.querySingle
+import org.jetbrains.kotlin.cli.common.CLIConfigurationKeys
+import org.jetbrains.kotlin.cli.jvm.config.JavaSourceRoot
+import java.io.File
+
+internal class KotlinAnalysisSourceRootsExtractor : SourceRootsExtractor {
+    override fun extract(sourceSet: DokkaConfiguration.DokkaSourceSet, context: DokkaContext): List<File> {
```

Comment:
It might indeed be so and it might be possible to simplify it, but seeing how this function is invoked from other parts of the compiler (see the screenshot) - that's the type of change I don't feel comfortable doing at this moment, and I think it's much safer to preserve the original behaviour, even if it adds an additional internal class.

![2023-07-03_15-22-31](https://github.com/Kotlin/dokka/assets/22685910/782e5b8d-778f-4813-8ea3-d3460757b5f7)

Unfortunately, I had already "fixed" a few much simpler cases where I thought everything was  clear, and then spent half a day debugging what went wrong :) For instance, [removing two unused variables](https://github.com/Kotlin/dokka/blob/98160128870a1de3a2a3319c51b6715a0a2b64c7/subprojects/analysis-kotlin-descriptors/compiler/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/descriptors/compiler/configuration/AnalysisEnvironment.kt#L263-L269) broke ~15 out of 700 unit tests, and it wasn't obvious what went wrong